query/filter/selector.go

```diff
@@ -144,7 +145,17 @@ func (s *Selector) ToSearchFilter() []string {
 }
 
 func (s *Selector) IsIndexed() bool {
-	return !(s.Field.DataType == schema.ByteType || s.Matcher.GetValue().AsInterface() == nil)
+	switch {
+	case s.Field.DataType == schema.DoubleType:
+		v := s.Matcher.GetValue().(*value.DoubleValue).Double
```

Comment:
It's better to protect type assertion.

Reply:
the case is first checking if it is double, the building of value guarantees this as value is built from field type.

Reply:
Linters would still complain.

Reply:
hmm